### PR TITLE
Add a count method to the OrganizationMembersResource

### LINKED ISSUE #31388: [Organizations] Add a count() method to the OrganizationMembersResource
### Description

In some use cases, knowing the amount of users that are members of an organization without having to query them all is a desired feature. Currently, the `OrganizationMembersResource` endpoint offers no method for this, and adding one should be a simple task given that in Keycloak we already have queries in place to count all users that are members of a group.

### Discussion

_No response_

### Motivation

Main motivation is to unlock more flexibility for users of the API to be able to get the number of org members without having to query all members.

### Details

_No response_

### PR DESCRIPTION
**Pull request by martin-kanis**
Closes #31388

### CODE REVIEW

**Review by sguilhen — COMMENTED** [review]
@martin-kanis LGTM, not sure why some tests are failing - the js ones don't seem to be related, were you able to check the model tests failures?

**Comment by martin-kanis** [comment]
@sguilhen Model tests are failing regularly recently https://github.com/keycloak/keycloak/issues/31612. Those are not related to this PR.

**Comment by martin-kanis** [comment]
@pedroigor @sguilhen GHA are finally green.

**Review by sguilhen — APPROVED** [review]

File: model/storage-private/src/main/java/org/keycloak/storage/UserStorageManager.java [hunk 1]
```diff
@@ -360,6 +360,8 @@ public boolean removeUser(RealmModel realm, UserModel user) {
             getFederatedStorage().preRemove(realm, user);
         }
 
+        publishUserPreRemovedEvent(realm, user);
```
Comment (pedroigor) [inline model/storage-private/src/main/java/org/keycloak/storage/UserStorageManager.java:363]: I think a better pattern is what we are using for group events. See https://github.com/keycloak/keycloak/blob/main/server-spi/src/main/java/org/keycloak/models/GroupModel.java#L67.

Not a blocker though as we can change that in a follow-up.

**Review by pedroigor — DISMISSED** [review]

**Review by mposolda — DISMISSED** [review]
Approving for core-clients (Relying mostly on existing review by @pedroigor and @sguilhen )

**Review by pedroigor — APPROVED** [review]

**Pull request merged**